src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs

```diff
@@ -206,7 +206,7 @@ public abstract partial class InMemoryDocumentSessionOperations : IDisposable
             MaxNumberOfRequestsPerSession = _requestExecutor.Conventions.MaxNumberOfRequestsPerSession;
             GenerateEntityIdOnTheClient = new GenerateEntityIdOnTheClient(_requestExecutor.Conventions, GenerateId);
             EntityToBlittable = new EntityToBlittable(this);
-            SessionInfo = new SessionInfo(_clientSessionId, false, _documentStore.LastTransactionIndex);
+            SessionInfo = new SessionInfo(_clientSessionId, false, () => _documentStore.GetLastTransactionIndex(DatabaseName));
```

Comment:
Let's avoid the lamda here and just return the current value

Reply:
The problem is, if we do several save changes in the same session.

Reply:
See later, your usage of this in RequestExecuter isn't good